hack/dind-cluster.sh

```diff
@@ -71,175 +57,74 @@ set -o errexit
-  sudo modprobe br_netfilter 2> /dev/null || true
-  sudo sysctl -w net.bridge.bridge-nf-call-iptables=0 > /dev/null
-  # overlayfs, if available, will be faster than vfs
-  sudo modprobe overlay 2> /dev/null || true
-  mkdir -p "${CONFIG_ROOT}"
-
-  if [[ "${SKIP_BUILD}" = "true" ]]; then
-    echo "WARNING: Skipping image build due to OPENSHIFT_SKIP_BUILD=true"
-  else
-    build-images
-  fi
+  echo "Starting dind cluster '${cluster_id}' with plugin '${network_plugin}'"
 
-  ## Create containers
-  echo "Launching containers"
-  local root_volume="-v ${OS_ROOT}:${DEPLOYED_ROOT}"
-  local config_volume="-v ${CONFIG_ROOT}:${DEPLOYED_CONFIG_ROOT}"
-  local volumes="${root_volume} ${config_volume}"
-  # systemd requires RTMIN+3 to shutdown properly
-  local stop="--stop-signal=$(kill -l RTMIN+3)"
-  local base_run_cmd="${DOCKER_CMD} run -dt ${stop} ${volumes}"
-
-  local master_cid="$(${base_run_cmd} --privileged --name="${MASTER_NAME}" \
-      --hostname="${MASTER_NAME}" "${DIND_IMAGE}")"
-  local master_ip="$(get-docker-ip "${master_cid}")"
-
-  local node_cids=()
-  local node_ips=()
+  # Ensuring compatible host configuration
+  #
+  # Running in a container ensures that the docker host will be affected even
+  # if docker is running remotely.  The openshift/dind-node image was chosen
+  # due to its having sysctl installed.
+  ${DOCKER_CMD} run --privileged --net=host --rm -v /lib/modules:/lib/modules \
+                openshift/dind-node bash -e -c \
+                '/usr/sbin/sysctl -w net.bridge.bridge-nf-call-iptables=0 > /dev/null;
+                modprobe openvswitch;
+                modprobe overlay 2> /dev/null || true;
+                modprobe br_netfilter 2> /dev/null || true;'
+
+  # Initialize the cluster config path
+  mkdir -p "${config_root}"
+  echo "${network_plugin}" > "${config_root}/network-plugin"
+  copy-runtime "${origin_root}" "${config_root}/"
+
+  local volumes="-v ${config_root}:${deployed_config_root}"
+  local run_cmd="${DOCKER_CMD} run -dt ${volumes}  --privileged"
+
+  # Create containers
+  local master_cid
+  master_cid="$(${run_cmd} --name="${MASTER_NAME}" --hostname="${MASTER_NAME}" "${MASTER_IMAGE}")"
+  local master_ip
+  master_ip="$(get-docker-ip "${master_cid}")"
+  local entry="${master_ip}\t${MASTER_NAME}-node"
+  ${DOCKER_CMD} exec -t "${MASTER_NAME}" bash -c "echo -e \"${entry}\" >> /etc/hosts"
   for name in "${NODE_NAMES[@]}"; do
-    local cid="$(${base_run_cmd} --privileged --name="${name}" \
-        --hostname="${name}" "${DIND_IMAGE}")"
-    node_cids+=( "${cid}" )
-    node_ips+=( "$(get-docker-ip "${cid}")" )
+    local cid
+    cid="$(${run_cmd} --name="${name}" --hostname="${name}" "${NODE_IMAGE}")"
+    local ip
+    ip="$(get-docker-ip "${cid}")"
+    entry="${ip}\t${name}"
+    # Ensure the master can resolve node names to ip for kubelet communication
+    #
+    # Attempts to keep /etc/hosts in sync with the api's node records
+    # have thus far proved unreliable.
```

Comment:
I was having trouble naively using a systemd unit on a timer.  Now that @eparis pointed me at @smarterclayton's guide to using ``oc observe``, I think a bash control loop might do the trick.